quickwit-cli/tests/cli.rs

```diff
@@ -225,6 +234,61 @@ fn test_cmd_delete() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_cmd_garbae_collect() -> Result<()> {
+    let test_env = create_test_env(TestStorageType::LocalFileSystem)?;
+    create_logs_index(&test_env);
+    index_data(
+        &test_env.index_uri,
+        test_env.resource_files["logs"].as_path(),
+    );
+
+    let (metastore_uri, index_id) =
+        extract_metastore_uri_and_index_id_from_index_uri(&test_env.index_uri)?;
+    let metastore = MetastoreUriResolver::default()
+        .resolve(metastore_uri)
+        .await?;
+    let splits = metastore.list_all_splits(index_id).await?;
+    assert_eq!(splits.len(), 1);
+    make_command(format!("gc --index-uri {}", test_env.index_uri).as_str())
+        .assert()
+        .success()
+        .stdout(predicate::str::contains(
+            "No dangling files to garbage collect",
+        ));
+
+    let split_ids = vec![splits[0].split_id.as_str()];
+    metastore
+        .mark_splits_as_deleted(index_id, split_ids)
+        .await?;
+    make_command(format!("gc --index-uri {} --dry-run", test_env.index_uri).as_str())
+        .assert()
+        .success()
+        .stdout(predicate::str::contains(
+            "The following files will be garbage collected.",
+        ))
+        .stdout(predicate::str::contains("/hotcache"))
+        .stdout(predicate::str::contains("/.manifest"));
```

Comment:
This is just checking if the `dry-run` listing contains few files we know exist in the split.
we could construct the list of split files since we know the split_id but I was keeping it simple.
The split deletion is & checking happens in the next lines. I will add a check to make sure the split folder is actually deleted.